src/viam/trajex/totg/trajectory.cpp

```diff
@@ -937,6 +937,11 @@ trajectory trajectory::create(class path p, options opt, integration_points poin
 
         // State transition function - validates and performs state transitions based on events
         const auto transition = [](integration_state current_state, integration_event event) -> integration_state {
+            // Handle self-transition uniformly for all states
+            if (event == integration_event::k_none) [[likely]] {
```

Comment:
It was weird that this function had no handling for k_none, which is the most common event. Now it handles it, which lets us just call `transition` with any event, and not need to special case not call it for `k_none`.